code-challenges/value-vs-reference/challenges-02.test.js

```diff
@@ -0,0 +1,166 @@
+'use strict';
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1
+
+Write a function that appends ' The end.' to a string, and returns the modified string. The original source string should not be modified.
+
+------------------------------------------------------------------------------------------------ */
+
+const appendTheEnd = (str) => {
+  // Solution code here...
+  let a = str;
+  let b = `${str} The end.`
+  return a,b
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function that accepts an array and copies the first element to the end of the array. The change should be reflected in the source array that was passed in to the function. That is, the function should modify the array 'in place'.
+
+Do not use a return statement.
+
+For example:
+const a = [1, 2, 3];
+appendFirstToLast(a);
+console.log(a) prints [1, 2, 3, 1]
+------------------------------------------------------------------------------------------------ */
+
```

Comment:
the goal of this is to make you code dynamic, you want to make this function work for any array given, not just the one written in the test.